### PR TITLE
Fix `esasky` remote tests

### PR DESCRIPTION
**Pull request by eerovaher**
The two reasons the tests were failing were not using uppercase filenames for file existence checks where needed and not closing HDULists.

Contributes towards #2634

### CODE REVIEW

File: astroquery/esasky/tests/test_esasky_remote.py [hunk 1]
```diff
@@ -115,10 +118,15 @@ def test_esasky_query_object_maps(self):
                                          'ISO-IR', 'Herschel', 'JWST-MID-IR',
                                          'JWST-NEAR-IR', 'Spitzer'])
     def test_esasky_get_images(self, tmp_path, mission):
-        ESASky.get_images(position="M51", missions=mission, download_dir=tmp_path)
-
+        result = ESASky.get_images(
+            position="M51", missions=mission, download_dir=tmp_path
+        )
```
Comment (bsipocz) [inline astroquery/esasky/tests/test_esasky_remote.py:123]: keep these long, we use line length 120.

File: astroquery/esasky/tests/test_esasky_remote.py [hunk 2]
```diff
@@ -78,13 +78,15 @@ def test_esasky_get_images_obs_id(self, tmp_path, mission, obsid):
...
-        assert Path(tmp_path, mission).exists()
+        assert Path(tmp_path, mission.upper()).exists()
         if mission == "Herschel":
             assert isinstance(result[mission.upper()][0]["250"], HDUList)
             assert isinstance(result[mission.upper()][0]["350"], HDUList)
             assert isinstance(result[mission.upper()][0]["500"], HDUList)
         else:
             assert isinstance(result[mission.upper()][0], HDUList)
+            for hdu_list in result[mission.upper()]:
+                hdu_list.close()
```
Comment (bsipocz) [inline astroquery/esasky/tests/test_esasky_remote.py:89]: Thanks.  I wonder whether this should rather be dealt with by a context manager somewhere in the module instead, but I suppose getting a workaround in the test would do for now.

File: astroquery/esasky/tests/test_esasky_remote.py (hunk 1 above)
Comment (eerovaher) [reply astroquery/esasky/tests/test_esasky_remote.py]: I've merged the line triplets.

**Review by bsipocz — APPROVED** [review]

**Comment by bsipocz** [comment]
Thanks @eerovaher!

**Pull request merged**